app/utils/service_client/middleware/jwt_authenticator.rb

```diff
@@ -0,0 +1,25 @@
+module ServiceClient
+  module Middleware
+
+    TOKEN_DATA_KEYS = Set.new([:marketplaceId, :actorId, :role])
+
+    class JwtAuthenticator < MiddlewareBase
+      def enter(ctx)
+        unless APP_CONFIG.harmony_api_disable_authentiation
```

Comment:
Instead of accessing the `APP_CONFIG` global, I'd recommend to add a `.initialize` method and pass the value as a constructor parameter. Makes testing much easier.